### PR TITLE
Make top level section errors more prominent / Allow global keywords before RUNSPEC

### PR DESCRIPTION
**Pull request by blattms**
If there are errors at the top level (e.g. RUNSPEC comes after NOECHO), we might get very many errors and the user might miss the relevant ones on top.

To fix this we will now print those errors last just before exiting the simulator because of these errors.

### CODE REVIEW

**Comment by bska** [comment]
Very, very strictly speaking, there is a small selection of keywords which are meaningful before `RUNSPEC`.  `ECHO`, `END`, `INCLUDE`, and `NOECHO` are probably the most important here.  We don't currently support `ECHO`/`NOECHO`, but on the other hand I think it might be helpful to ignore those two keywords in this particular context&ndash;before `RUNSPEC`&ndash;if at all possible.

**Comment by blattms** [comment]
jenkins build this please

**Comment by blattms** [comment]
jenkins build this please

**Comment by blattms** [comment]
And now it should finally work :sweat_smile:

**Comment by blattms** [comment]
jenkins build this please

**Review by bska — COMMENTED** [review]
This looks good to me except for a tiny inconsistency in the documentation of the new helper function.  I'm happy to merge this in its current state, but if you'd like to fix the documentation then please go ahead.

File: src/opm/input/eclipse/Parser/Parser.cpp [hunk 1]
```diff
@@ -74,6 +74,22 @@
 #include <fmt/format.h>
 
 namespace {
+    /// \brief Whether a keyword is a global keyword and how many to skip
+    ///
+    /// Those are allowed before RUNSPEC.
+    /// \return false if this is not a global keyword or the number of keywords to skip
+    ///               otherwise (including itself.
```
Comment (bska) [inline src/opm/input/eclipse/Parser/Parser.cpp:81]: The documentation says that the function returns false or a number but it only returns `true` or `false`.

Comment (blattms) [reply src/opm/input/eclipse/Parser/Parser.cpp]: Argh! Thanks for noticing. Fixed

**Review by bska — APPROVED** [review]
Thanks a lot for the updates.  This looks good to me now and I'll merge into master.

**Pull request merged**